### ISSUE TITLE
Comparison with Timenet

### ISSUE THREAD

**Issue by ksachdeva** [White-Link/UnsupervisedScalableRepresentationLearningTimeSeries#24]
Hi,

I found the reference and comparison to the Timenet paper

Malhotra, P., TV, V., Vig, L., Agarwal, P., and Shroff, G. TimeNet: Pre-trained deep recurrent neural network for time series classification.

and I found the way they construct their dataset for training, validation & testing is different from yours.

In your experiments, you train an encoder only on **one** dataset at a time and use it to generate the representations for any other dataset.

In Timenet, they selected 18 datasets to be used for training. Here is a paragraph from their paper

> The training dataset is diverse as it contains time series belonging to 151 different classes from the 18 datasets with T varying from 24 to 512 (refer Table 2 for details). When using a dataset as part of the training or validation set, all the train and test time series from the dataset are included in the set. Each time series is normalized to have zero mean and unit variance

Unless I have misunderstood your paper & code, the size of the training dataset for your encoder(s) would be far less than the ones they used. Please correct me if my interpretation is incorrect.

If my interpretation is correct, do you think having a training setup like Timenet is of advantage to your network architecture & loss function as well?

Regards & thanks
Kapil

**Comment by White-Link** [comment]
Hi, thank for your question that enables us to clarify this point of the paper.

We actually identified the issue that you raise, and chose instead to compare our method against the "SAE-C" variant (see Section 4.3) proposed by Malhotra et al. as it corresponds to the TimeNet model but trained only one dataset at a time, which corresponds to our setting. We are sorry about the confusing wording that we used in our paper.

We did not experiment training on multiple dataset to potentially improve the performance of our model, but I assume this would help, as hinted by Malhotra et al., although I am unsure of the extent of the improvement.

**Comment by ksachdeva** [comment]
Many thanks for this pointer. It makes sense now.

>We did not experiment training on multiple dataset to potentially improve the performance of our model, but I assume this would help, as hinted by Malhotra et al., although I am unsure of the extent of the improvement.

When I looked at their training setup, initially I was a bit surprised. I would have assumed that time series from different datasets belong to different domains and if it is even valid to combine them. But then, their experiment is proving its utility.  Would appreciate it if you have any opinion/insights on this.

Regards
Kapil

**Comment by White-Link** [comment]
Hi, sorry for the late reply! Unfortunately, we have no other experimental result to share on this matter.

I would say that this combination of multiple datasets might not be as effective for our method, since the proposed time-based sampling in our paper actually amounts to increasing the size of the training dataset by subsampling time series, but could be useful to improve performance, especially when dealing with small datasets, or groups of similar datasets (such as CricketX, CricketY, CricketZ in the UCR archive).

Nonetheless, training on multiple datasets like TimeNet brings interesting generalization properties as it yields a representation model that is applicable without further training to numerous other datasets, the learned representations being general thanks to the training setting.
